### PR TITLE
Remove ExtenedMap, ToggleSet, and observable versions

### PR DESCRIPTION
**Pull request by Nokel81**
- Switch to helper functions instead. They don't impact the prototype
  tree and it means that we don't need to introduce versions that are
  both observable and "normal"

- Tidy <ItemListLayout> and hiddenTableColumns to not need
  pre-initialization

Signed-off-by: Sebastian Malton <sebastian@malton.name>

### CODE REVIEW

File: src/renderer/components/item-object-list/item-list-layout.tsx [hunk 1]
```diff
@@ -145,10 +145,6 @@ export class ItemListLayout<I extends ItemObject> extends React.Component<ItemLi
       throw new Error("[ItemListLayout]: configurable list require props.tableId to be specified");
     }
 
-    if (isConfigurable && !UserStore.getInstance().hiddenTableColumns.has(tableId)) {
-      UserStore.getInstance().hiddenTableColumns.set(tableId, new ObservableToggleSet());
-    }
-
```
Comment (jim-docker) [inline src/renderer/components/item-object-list/item-list-layout.tsx]: this is getting done now elsewhere?

**Review by jim-docker — COMMENTED** [review]
do we really need this for 5.3?

File: src/renderer/components/item-object-list/item-list-layout.tsx (hunk 1 above)
Comment (Nokel81) [reply src/renderer/components/item-object-list/item-list-layout.tsx]: Yes, it is being done https://github.com/lensapp/lens/pull/3310/files/97776b9c48b8e549ea5390c3bd4cd1833a7d43a2#diff-dcf7a988ad98b11a4537826fdb986a5ef56f35bdc827e66285c57138446caacaR174-R177

**Comment by Nokel81** [comment]
@jim-docker Probably not, I wanted to use this in a different PR... but I will move this to 5.3.1 instead.

**Comment by jim-docker** [comment]
Seeing this, though probably not related to changes in this PR?

Screen Shot 2022-02-14 at 3 43 21 PM

**Comment by Nokel81** [comment]
> Seeing this, though probably not related to changes in this PR?

Did you notice what the console error was too?

**Comment by Nokel81** [comment]
That sounds like you unloaded the lenscloud extension but were still viewing the ManagedDevCluster category

**Comment by Nokel81** [comment]
So yes, not related to this PR.

**Comment by Nokel81** [comment]
@jim-docker PTALA

**Review by aleksfront — DISMISSED** [review]
LGTM

**Review by aleksfront — APPROVED** [review]

**Review by jim-docker — APPROVED** [review]

**Pull request merged**